ocrd_models/ocrd_page_user_methods/get_AllRegions.py

```diff
@@ -0,0 +1,74 @@
+# pylint: disable=line-too-long,invalid-name,protected-access,missing-module-docstring
+def _region_class(self, x): # pylint: disable=unused-argument
+    return x.__class__.__name__.replace('RegionType', '')
+
+def _get_recursive_regions(self, regions, level, classes=None):
+    if level == 1:
+        # stop recursion, filter classes
+        if classes:
+            return [r for r in regions if self._region_class(r) in classes]
+        # remove the first element (PageType)
+        return list(set(regions[1:]))
```

Comment:
What's the reason for this `list(set())` construct? It can't be to achieve the effect of `.copy()`, so I am guessing you want `set` to re-order by hash/ID?

Reply:
To remove duplicates, without it, I get some ~1000 regions for depth 10, ~1500 for depth 11 etc.

Reply:
Oh, right. But then the recursion itself is wrong. Let's change the loop such that the new regions are always derived from the _previous_ new regions only.

Reply:
Yes, I'm fixing it now.